### PR TITLE
SVS-106: Customize the name field of the RuleSet file

### PR DESCRIPTION
**Pull request by Evangelink**

### CODE REVIEW

File: src/Integration.UnitTests/Binding/BindingWorkflowTests.cs [hunk 1]
```diff
@@ -81,6 +81,7 @@ public void BindingWorkflow_DownloadQualityProfile_Success()
             var notifications = new ConfigurableProgressStepExecutionEvents();
 
             RuleSet ruleSet = TestRuleSetHelper.CreateTestRuleSetWithRuleIds(new[] { "Key1", "Key2" });
+            var expectedRuleSet = new RuleSet(ruleSet) { NonLocalizedDisplayName = "SonarQube -  ", NonLocalizedDescription = "\r\nhttp://connected/profiles/show?key=" };
```
Comment (tvsonar) [inline src/Integration.UnitTests/Binding/BindingWorkflowTests.cs]: This expected ruleset name looks strange with the double spaces.

File: src/Integration.Vsix/Integration.Vsix.csproj [hunk 2]
```diff
@@ -33,7 +33,7 @@
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
     <DeployExtension>True</DeployExtension>
-    <CodeAnalysisRuleSet>Integration.Vsix.ruleset</CodeAnalysisRuleSet>
+    <CodeAnalysisRuleSet>..\Integration.Vsix.UnitTests\Integration.Vsix.UnitTests.ruleset</CodeAnalysisRuleSet>
```
Comment (tvsonar) [inline src/Integration.Vsix/Integration.Vsix.csproj]: This change is not relevant for this PR. Also, it's strange to reference a unittest ruleset in a non-unit test project.

File: src/Integration/Binding/BindingWorkflow.cs [hunk 3]
```diff
@@ -229,6 +229,9 @@ private ProgressStepDefinition[] CreateWorkflowSteps(IProgressController control
                 File.WriteAllText(tempRuleSetFilePath, export.Configuration.RuleSet.OuterXml);
                 RuleSet ruleSet = RuleSet.LoadFromFile(tempRuleSetFilePath);
 
+                // Remove/Move/Refactor me when we no longer download a XML file but actually use the proper API
```
Comment (tvsonar) [inline src/Integration/Binding/BindingWorkflow.cs]: Let's change comment to: `Remove/Move/Refactor code when XML ruleset file is no longer downloaded but the proper API is used to retrieve rules`

File: src/TestInfrastructure/Framework/ConfigurableSolutionRuleStore.cs [hunk 4]
```diff
@@ -39,8 +43,12 @@ void ISolutionRuleStore.RegisterKnownRuleSets(IDictionary<Language, RuleSet> rul
 
         public void RegisterRuleSetPath(Language language, string path)
         {
-            this.registeredPaths[language] = path;
+            if (!this.availableRuleSets.ContainsKey(language))
```
Comment (tvsonar) [inline src/TestInfrastructure/Framework/ConfigurableSolutionRuleStore.cs]: Please always use curly braces.

Comment (tvsonar) [inline src/TestInfrastructure/Framework/ConfigurableSolutionRuleStore.cs]: Note to self: enable curly brace rule on Dory.

File: src/Integration/Resources/Strings.resx [hunk 5]
```diff
@@ -660,4 +660,20 @@ This is the general format to use when writing errors to output window or when t
     <value>   {0}</value>
     <comment>Output window message format. Used whenever the message should be displayed as a sub-message from a step message so the user can clearly identify the grouping. {0} the actual message to display.</comment>
   </data>
+  <data name="SonarQubeProjectSolutionWideRuleSetPrefix" xml:space="preserve">
+    <value>SonarQube</value>
+    <comment>Used in combination with SonarQubeRuleSetNameFormat when the ruleset is the solution ruleset</comment>
+  </data>
+  <data name="SonarQubeProjectSpecificRuleSetPrefix" xml:space="preserve">
+    <value>Local</value>
```
Comment (tvsonar) [inline src/Integration/Resources/Strings.resx]: Let's use SonarQube here as well. So project specific and solution level ruleset files will have the same name.

File: src/Integration/Binding/BindingWorkflow.cs (hunk 3 above)
Comment (Evangelink) [inline src/Integration/Binding/BindingWorkflow.cs]: Done

File: src/Integration.Vsix/Integration.Vsix.csproj (hunk 2 above)
Comment (Evangelink) [inline src/Integration.Vsix/Integration.Vsix.csproj]: Done

File: src/TestInfrastructure/Framework/ConfigurableSolutionRuleStore.cs (hunk 4 above)
Comment (Evangelink) [inline src/TestInfrastructure/Framework/ConfigurableSolutionRuleStore.cs]: Done

File: src/Integration/Binding/ISolutionRuleStore.cs [hunk 6]
```diff
@@ -22,8 +22,8 @@ internal interface ISolutionRuleStore
         void RegisterKnownRuleSets(IDictionary<Language, RuleSet> ruleSets);
 
         /// <summary>
-        /// Retrieves the path for the solution-level <see cref="RuleSet"/> mapped to the <see cref="Language"/>.
+        /// Retrieves the solution-level <see cref="RuleSet"/> mapped to the <see cref="Language"/>.
         /// </summary>
-        string GetRuleSetFilePath(Language language);
+        SolutionBindingOperation.RuleSetInformation GetRuleSetInformation(Language language);
```
Comment (Evangelink) [inline src/Integration/Binding/ISolutionRuleStore.cs]: Done

File: src/Integration.UnitTests/Binding/ProjectBindingOperation.WriterTests.cs [hunk 7]
```diff
@@ -375,7 +390,8 @@ public void ProjectBindingOperation_QueueWriteProjectLevelRuleSet_NewBinding()
             // Act (write pending)
             this.sccFileSystem.WritePendingNoErrorsExpected();
 
-                // Verify
+            // Verify
+            expectedRuleSet.NonLocalizedDisplayName = "Local";
```
Comment (Evangelink) [inline src/Integration.UnitTests/Binding/ProjectBindingOperation.WriterTests.cs]: Done

**Pull request merged**